src/compiler/checker.ts

```diff
@@ -38325,6 +38331,9 @@ namespace ts {
                     const typeNode = getEffectiveTypeAnnotationNode(getRootDeclaration(declaration));
                     if (typeNode) {
                         const type = getTypeForVariableLikeDeclaration(declaration, /*includeOptionality*/ false);
+                        if (isObjectBindingPattern(declaration.name) && type === unknownType) {
+                            forEach(declaration.name.elements, checkSourceElement);
+                        }

```

Comment:
I think it'd be better to stick `checkVariableLikeDeclaration(declaration)` right below its declaration above; I don't know if there's predicant for conditionally walking like this based on type.

Doing that `checkVariableLikeDeclaration` does introduce a load of "implicit any" errors, which I _think_ are arguably correct, but would probably be a sweeping change.

Pending that, I don't think that this `forEach` is correct; it seems like it doesn't recurse down when it should for consistency (see `destructuringCatch`'s examples with `catch ([{x: [y], z}])` and such). That test also poses an interesting question about the behavior of something like `catch ([a, b])`, though.

Reply:
@jakebailey I've converted the PR into a project. I'll check it later when I have time.

Reply:
No worries; I really appreciate you working on this.

Just to note it here before I forget, if I'm remembering correctly, errors about the "same thing" are to be avoided; [my example](https://www.typescriptlang.org/play?#code/MYewdgzgLgBA3jAHgTwF4wL4C4YFcwDWYIA7mDALzwYDcAUKJCADYCmAdMyAOYAUKqAJT06AM3zAoAS3AxRIELwRgAhgFtWmHPiKkwg+HRjGYjCCw5c+qjcLoY6dKACdkhkzAD0nmOz-2GFShgAAsYJRgbTWw8QmIyAzgjEzMLTh5eKLsMIA) shows that the existing destructuring of `unknown` don't issue `implict any`, so there's something to look into here about how those are walked and how they avoid that double error.